BOARD: Add Ethernet support to frdm imx93

CkovMk opened the pull request:
Add YT8521 PHY driver and enable it on FRDM-IMX93

whbo158 committed:
dts: arm64: nxp: frdm_imx93: enable enet support

PHY chip YT8521, the driver is:
drivers/ethernet/phy/phy_motorcomm_yt8521.c

Signed-off-by: Hongbo Wang <hongbo.wang@nxp.com>

whbo158 committed:
drivers: eth: phy: add Motorcomm YT8521 PHY driver

Test PHY driver using zperf:
west build -p always -b frdm_imx93/mimx9352/a55 samples/net/zperf/

Signed-off-by: Hongbo Wang <hongbo.wang@nxp.com>

maass-hamburg commented:
Why can't the generic `ethernet-phy` already in zephyr can't be used instead. I don't see you using external irq pins, that would indicate the need for a special driver.

CkovMk committed:
drivers: update motorcomm yt8521 driver (squash before merge)

- Moves binding to phy subdir
- Add rx and tx delay sel
- Format code
- Update frdm-imx93 dts accordingly

Signed-off-by: Chekhov Ma <chekhov.ma@nxp.com>

CkovMk commented on drivers/ethernet/phy/phy_motorcomm_yt8521.c:
```diff
@@ -0,0 +1,623 @@
+		return -EIO;
+	}
+
+	/* set default reg space */
+	mc_ytphy_write_ext(dev, YT8521_REG_SPACE_SELECT_REG, YT8521_RSSR_UTP_SPACE);
+
+	mc_ytphy_modify_ext(dev, YTPHY_SYNCE_CFG_REG, YT8521_SCR_SYNCE_ENABLE, 0);
+
+	ret = mc_ytphy_get_features(dev);
+	if (ret) {
+		return -EIO;
+	}
+
+	/* Reset PHY */
+	ret = mc_ytphy_soft_reset(dev);
+	if (ret) {
+		return -EIO;
+	}
+
+	/* Enable clock delay */
+	mc_ytphy_cfg_clock_delay(dev);
+
+	mc_ytphy_resume(dev);
```
Added dt props

CkovMk commented:
Added rx and tx delay sel dt props.

Separate commit is created for easier review. Will squash it before merge.

CkovMk commented:
The driver needs to be adapted to recent Zephyr version. Convert to draft.
